made headings actual markdown headings

arne-broering opened the pull request:
I did this so that we can have anchor links to headings

### Description

Please provide a concise summary of the changes and their motivation.

#### Issues Addressed

List and link all the issues addressed by this PR.

#### Change Type

Please select the relevant options:

- [ ] Fix (change that resolves an issue)
- [ ] New enhancement (change that adds specification content)

#### Checklist

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document.
- [ ] My changes adhere to the established patterns, and best practices.

phil-abb commented:
@arne-broering I believe @ajcraig intentionally changed these to not be headers (and other places) before we moved to this repo because it was causing the sidebar navigation to be too deep. Not having them as headers breaks the links so I'm not sure what is better. 

@ajcraig What do you think?

phil-abb commented on system-design/app-interoperability/application-package-definition.md:
```diff
@@ -132,7 +132,7 @@ metadata:
       - name: Northstar Industrial Applications
         site: http://northstar-ida.com
-deploymentProfiles:
+s:
```
It looks like you may have accidentally deleted some text here.

arne-broering replied:
Just fixed that

arne-broering commented:
> @arne-broering I believe @ajcraig intentionally changed these to not be headers (and other places) before we moved to this repo because it was causing the sidebar navigation to be too deep. Not having them as headers breaks the links so I'm not sure what is better.
> 
> @ajcraig What do you think?

Ah, ok, didn't know that. I just changed those to actual "#" headings, as it would be really helpful to link in the type definitions from the Deployment spec to the App Package spec.

arne-broering commented:
I added the following lines to `mkdocs.yml`:

```
markdown_extensions:
  - toc:
      toc_depth: 2
```
